### PR TITLE
added email button render check

### PR DESCRIPTION
**Pull request by memurats**
The email header menu item should not be rendered on public pages and only for logged in users, therefore:

Added listener to check if all requirements are met to allow the render script to be included before the template is rendered

Cleaned up Application class

### CODE REVIEW

File: lib/Listener/BeforeTemplateRenderedListener.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	public function __construct(IUserSession $userSession) {
+		$this->userSession = $userSession;
+	}
+
+	public function handle(Event $event): void {
+		$user = $this->userSession->getUser();
+
+		if ($user === null) {
+			return;
+		}
```
Comment (juliushaertl) [inline lib/Listener/BeforeTemplateRenderedListener.php:32]: Nitpick: This is kind of duplicate with the `isLoggedIn()` return value of the event

https://github.com/nextcloud/server/blob/f3ae617bde99f8babddca634edb1f1f632b69191/lib/private/AppFramework/Middleware/AdditionalScriptsMiddleware.php#L52
https://github.com/nextcloud/server/blob/master/lib/private/User/Session.php#L269-L275

**Review by juliushaertl — APPROVED** [review]

File: lib/Listener/BeforeTemplateRenderedListener.php (hunk 1 above)
Comment (memurats) [reply lib/Listener/BeforeTemplateRenderedListener.php]: Hi! You're absolutely right. I removed the redundancy.

**Comment by memurats** [comment]
Any hints regarding the error in Psalm? Not sure what's supposedly wrong.

ERROR: MissingTemplateParam - lib/Listener/BeforeTemplateRenderedListener.php:19:49 - OCA\NmcSpica\Listener\BeforeTemplateRenderedListener has missing template params when extending OCP\EventDispatcher\IEventListener, expecting 1 (see https://psalm.dev/182)

File: lib/Listener/BeforeTemplateRenderedListener.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\NmcSpica\Listener;
+
+use OCP\AppFramework\Http\Events\BeforeTemplateRenderedEvent;
+use OCP\AppFramework\Http\TemplateResponse;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+
+class BeforeTemplateRenderedListener implements IEventListener {
```
Comment (juliushaertl) [inline lib/Listener/BeforeTemplateRenderedListener.php:19]:
```suggestion
/** @template-implements IEventListener<Event|BeforeTemplateRenderedEvent> */
class BeforeTemplateRenderedListener implements IEventListener {
```

Should fix the psalm error

**Pull request merged**